eclipse-theia/theia
# #10026 Implement DocumentSymbolProviderMetadata

#### What it does


This PR adds the implementation for `DocumentSymbolProviderMetadata` and uses label in `MonacoOutlineContribution`:
- Introduce DocumentSymbolProviderMetadata interface in theia.d.ts
- Adapt register functions to pass on optional metadata
- Adapt the MonacoOutlineContribution and use metadata label or plugin's displayName as root node if multiple providers are registered (aligned to behavior in VSCode)
- Remove Theia's custom DocumentSymbolProvider for 'launch.json' files and reuse VSCode's provider (available via vscode-builtin-configuration-editing)

Contributed on behalf of STMicroelectronics

Signed-off-by: Nina Doschek <ndoschek@eclipsesource.com>

Fixes #10026

#### How to test


1. Open a `launch.json` file in your workspace (e.g. https://github.com/eclipse-theia/theia/blob/master/.vscode/launch.json)
2. Open the Outline View via menu `View` > `Outline`
3. Comparable to VSCode, there are two providers registered:
   1. The plugin `vscode-builtin-json-language-features`
       Provides the JSON document outline under the root node `JSON Language Features (built-in)`
   2. The plugin `vscode-builtin-configuration-editing`
       Provides all launch configs as document symbols under the root node `Launch Targets`.
4. It is shown that either [the metadata label (in case of 3.ii.)](https://github.com/microsoft/vscode/blob/0a644fc76517b1aeeb37b8fdf8d636d39a3d9474/extensions/configuration-editing/src/configurationEditingMain.ts#L109) is used as root node display name or if not available the plugin display name (in case of 3.i.).
      

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
